source/common/network/connection_impl.cc

```diff
@@ -588,6 +592,19 @@ void ConnectionImpl::onReadReady() {
     result.action_ = PostIoAction::Close;
   }
 
+  if (result.action_ == PostIoAction::CloseError) {
+    if (dynamic_cast<ServerConnectionImpl*>(this)) {
+      stream_info_.setConnectionTerminationDetails(kDownstreamConnectionTerminationDetails);
+      stream_info_.setResponseFlag(StreamInfo::ResponseFlag::DownstreamConnectionTermination);
+    } else {
+      stream_info_.setUpstreamTransportFailureReason(kUpstreamConnectionTerminationDetails);
+      stream_info_.setConnectionTerminationDetails(kUpstreamConnectionTerminationDetails);
+      stream_info_.setResponseFlag(StreamInfo::ResponseFlag::UpstreamConnectionTermination);
+    }
+    // Force "end_stream" so that filters can process this error.
+    result.end_stream_read_ = true;
```

Comment:
I usually associate end_stream_read_ with graceful termination.  Is it appropriate to set this?  Do we need alternate signaling to notify filters about abrupt terminations via either additional arguments to push pipeline or a new virtual method which is called on abrupt termination.

Reply:
I agree; I think of end_stream as always a graceful operation. I think a different signal for error is more appropriate.

Reply:
The issue is that there is different signal right now, and if we don't set this, then network filters won't be executed at all.

Note that this PR is meant to be a temporary fix until we have a proper solution for #13940, but that's going to require a lot more changes and time.

Reply:
This seems like a massive hack.  How much effort would it be to implement the real fix?

Reply:
That depends on what do you consider a real fix. I'm not too familiar with that part of the codebase, but if we want to propagate upstream connection events to network filters, then I imagine it's quite a lot of changes, so I think it would be better if one of maintainers could work on this.

Reply:
@ggreenway 

I don't know what to suggest.  It would be good to figure out how to do a catch-all cleanup of WASM resources on abnormal connection termination.  Could WASM detect connection termination based on an L4 filter that does the relevant signaling on destruction?

Reply:
FWIW, we have a workaround in #13836 that works for Wasm.